addons/purchase_stock/models/purchase.py

```diff
@@ -333,6 +346,20 @@ def write(self, values):
             self.filtered(lambda l: l.order_id.state == 'purchase')._create_or_update_picking()
         return result
 
+    def action_product_forecast_report(self):
+        self.ensure_one()
+        action = self.product_id.action_product_forecast_report()
+        action['context'] = {
+            'active_id': self.product_id.id,
+            'active_model': 'product.product',
+            'move_to_match_id': self.move_ids.filtered(lambda m: m.product_id == self.product_id).id,
```

Comment:
I think that it these case you should accept multiple values (`move_to_match_ids: [...]`), as discuss today the purchase order line can be link to multiple moves (e.g. with backorder) - (but it is ok for MO)